helm/sematic-server/templates/ingress.yaml

```diff
@@ -34,6 +35,26 @@ spec:
                 name: {{ $serviceName }}
                 port:
                   number: {{ $servicePort }}
+          {{- if $.Values.deployment.socket_io.dedicated }}
+          - path: {{ printf "%s/socket.io" (.path | trimSuffix "/") }}
```

Comment:
If the driver wants to access this path, it is like accessing it from outside the cluster; then, there is a round trip (egress + ingress) for the driver to access the socker.io pod. 

Why don't we directly let the driver pod access the Service? https://kubernetes.io/docs/tutorials/services/connect-applications-service/#accessing-the-service

Then we don't need this authoring here.

Reply:
that is not what you're looking at.  the file this code is located in `ingress.yaml` defines the configuration for the ingress itself.  i.e. it's telling the ingress which traffic should be sent to which internal service.

as for cluster-internal resources accessing the services directly, yes, this is exactly how it is already working by default, but we allow customers to configure this however they want.

Reply:
I just realized that this is to serve the case of *attached* case, when the driver has to access the socket io pod through the Ingress. So going through Ingress is unavoidable. And we have separate implementations for the `detached` case, which directly connects to socket io Service.

My concern is no more. I have here for you a suggestion for improvement. In that case, we don't have to define two ingress routing rules here, which do the same thing. If we can have the driver of the `attached` case also connect to `api/vi/events` path, with certain override, one rule is enough.  otherwise, it looks odd from a serving perspective why this Ingress wants to expose two paths doing the same thing. 

Even if decide to do this, there is no need to do in this PR.